tests/integration/fundBadERC20Redemption.test.js

```diff
@@ -0,0 +1,83 @@
+import { BN, toWei } from 'web3-utils';
+import { call, send } from '~/deploy/utils/deploy-contract';
+import { partialRedeploy } from '~/deploy/scripts/deploy-system';
+import { CONTRACT_NAMES } from '~/tests/utils/constants';
+import { setupFundWithParams } from '~/tests/utils/fund';
+import getAccounts from '~/deploy/utils/getAccounts';
+import { getFunctionSignature } from '~/tests/utils/metadata';
+
+let deployer, manager, investor;
+let defaultTxOpts, managerTxOpts, investorTxOpts;
+let contracts;
+let weth, omg, priceSource, zeroExExchange;
+let fund;
+let takeOrderSignature;
+let wethToEthRate, omgToEthRate;
+
+beforeAll(async () => {
+  [deployer, manager, investor] = await getAccounts();
+  defaultTxOpts = { from: deployer, gas: 8000000 };
+  managerTxOpts = { ...defaultTxOpts, from: manager };
+  investorTxOpts = { ...defaultTxOpts, from: investor };
+  const deployed = await partialRedeploy([CONTRACT_NAMES.FUND_FACTORY]);
+  contracts = deployed.contracts;
+
+  takeOrderSignature = getFunctionSignature(
+    CONTRACT_NAMES.ORDER_TAKER,
+    'takeOrder',
+  );
+
+  weth = contracts.WETH;
+  omg = contracts.OMG;
+  zeroExExchange = contracts.ZeroExV2Exchange;
+  priceSource = contracts.TestingPriceFeed;
+
+  const fundFactory = contracts.FundFactory;
+  const zeroExAdapter = contracts.ZeroExV2Adapter;
+
+  wethToEthRate = toWei('1', 'ether');
+  omgToEthRate = toWei('0.5', 'ether');
+
+  await send(
+    priceSource,
+    'update',
+    [
+      [weth.options.address, omg.options.address],
+      [wethToEthRate, omgToEthRate],
+    ],
+    defaultTxOpts
+  );
+
+  fund = await setupFundWithParams({
+    defaultTokens: [weth.options.address, omg.options.address],
+    exchanges: [zeroExExchange.options.address],
+    exchangeAdapters: [zeroExAdapter.options.address],
+    initialInvestment: {
+      contribAmount: toWei('1', 'ether'),
+      investor,
+      tokenContract: omg,
+    },
+    manager,
+    quoteToken: weth.options.address,
+    fundFactory
+  });
+});
+
+describe('Investor invests BadERC20 token to a fund', () => {
```

Comment:
Probably don't need a `describe` block, since it's only 1 test block below. You can remove this, or edit to "investor _redeems_ BadERC20 token _from_ a fund"